skbio/diversity/tests/test_block.py

```diff
@@ -0,0 +1,218 @@
+        dm3 = DistanceMatrix(np.array([[0, 1, 2, 3],
+                                       [1, 0, 4, 5],
+                                       [2, 4, 0, 6],
+                                       [3, 5, 6, 0]]), (0, 3, 4, 5))
+        exp = DistanceMatrix(np.array([[0, 0, 0, 1, 2, 3],
+                                       [0, 0, 0, 0, 0, 123],
+                                       [0, 0, 0, 0, 44, 0],
+                                       [1, 0, 0, 0, 64, 5],
+                                       [2, 0, 44, 64, 0, 6],
+                                       [3, 123, 0, 5, 6, 0]]), list(range(6)))
+
+        obs = _reduce([dm1, dm2, dm3])
+        npt.assert_equal(obs.data, exp.data)
+        self.assertEqual(obs.ids, exp.ids)
+
+    def test_block_beta_diversity(self):
+        exp = beta_diversity('unweighted_unifrac', self.table1, self.sids1,
+                             tree=self.tree1, otu_ids=self.oids1)
+        obs = block_beta_diversity('unweighted_unifrac', self.table1,
+                                   self.sids1, otu_ids=self.oids1,
+                                   tree=self.tree1, k=2)
+        npt.assert_equal(obs.data, exp.data)
+        self.assertEqual(obs.ids, exp.ids)
+
+    def test_generate_id_blocks(self):
+        ids = [1, 2, 3, 4, 5]
+        exp = [(np.array((0, 1)), np.array((0, 1))),
+               (np.array((0, 1)), np.array((2, 3))),
+               (np.array((0, 1)), np.array((4,))),
+               (np.array((2, 3)), np.array((2, 3))),
+               (np.array((2, 3)), np.array((4,))),
+               (np.array((4,)),   np.array((4,)))]
+
+        obs = list(_generate_id_blocks(ids, 2))
+        npt.assert_equal(obs, exp)
+
+    def test_block_party_notree(self):
+        counts = np.arange(15).reshape(5, 3)
+        exp = [{'counts': np.array([[0, 1, 2], [3, 4, 5]]),
+                'ids': np.array([0, 1])},
+               {'counts': np.array([[0, 1, 2], [3, 4, 5], [6, 7, 8],
+                                    [9, 10, 11]]),
+                'ids': np.array([0, 1, 2, 3])},
+               {'counts': np.array([[0, 1, 2], [3, 4, 5], [12, 13, 14]]),
+                'ids': np.array([0, 1, 4])},
+               {'counts': np.array([[6, 7, 8], [9, 10, 11]]),
+                'ids': np.array([2, 3])},
+               {'counts': np.array([[6, 7, 8], [9, 10, 11], [12, 13, 14]]),
+                'ids': np.array([2, 3, 4])},
+               {'counts': np.array([[12, 13, 14]]), 'ids': np.array([4])}]
+        obs = [_block_party(counts, rids, cids) for rids, cids in
+               _generate_id_blocks(list(range(5)), 2)]
+        npt.assert_equal(obs, exp)
+
+    def test_block_party_tree(self):
+        counts = np.array([[1, 1, 1],
+                           [1, 0, 1],
+                           [1, 0, 1],
+                           [0, 0, 1],
+                           [0, 1, 1]])
+        tree = TreeNode.read(io.StringIO('(a:1,b:2,c:3);'))
```

Comment:
Not important, but a shortcut is `TreeNode.read(['(a:1,b:2,c:3);'])`. (Lists of lines work in skbio as a "file" object).